integration/container/run_linux_test.go

```diff
@@ -401,3 +401,52 @@ func TestSeccomp(t *testing.T) {
 		}
 	}
 }
+
+func TestCgroupRW(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+
+	ctx := setupTest(t)
+	apiClient := testEnv.APIClient()
+
+	type testCase struct {
+		ops              []func(*container.TestContainerConfig)
+		expectedExitCode int
+	}
+	testCases := []testCase{
+		{
+			ops:              nil,
+			expectedExitCode: 0,
+		},
+		{
+			ops:              []func(*container.TestContainerConfig){container.WithSecurityOpt("writable-cgroups=true")},
+			expectedExitCode: 0,
+		},
+		{
+			// misspelled writable
+			ops:              []func(*container.TestContainerConfig){container.WithSecurityOpt("writeable-cgroups=true")},
+			expectedExitCode: 125,
+		},
+		{
+			ops:              []func(*container.TestContainerConfig){container.WithSecurityOpt("writable-cgroups=false")},
+			expectedExitCode: 0,
+		},
+		{
+			ops:              []func(*container.TestContainerConfig){container.WithSecurityOpt("writable-cgroups=1")},
+			expectedExitCode: 1,
+		},
+		{
+			ops:              []func(*container.TestContainerConfig){container.WithSecurityOpt("writable-cgroups=potato")},
+			expectedExitCode: 1,
+		},
+	}
+	for _, tc := range testCases {
+		cID := container.Run(ctx, t, apiClient, tc.ops...)
+		res, err := container.Exec(ctx, apiClient, cID, []string{"/bin/true"})
+		assert.NilError(t, err)
```

Comment:
This assert is failing for negative test cases.
Maybe only do this assert if `tc.expectedExitCode == 0`

Reply:
ok. I went about it in a slightly deconstructed way.